### ISSUE TITLE
NVDA not presenting emojis when writing them

### ISSUE THREAD

**Issue by andrew-l-d** [nvaccess/nvda#9120]
Using NVDA 2018.4 on Windows 10 1803.  Emojis not presented correctly under the following circumstances.
When writing in either Word or an Outlook email, if I include an emoji it is reported  as a number.  For example, the smiling eyes emoji when entered either by using colon and right parenthesis or by selecting from the Windows selection panel is announced as 12.  If I restart NVDA while in the document or email, the emoji is described.  If I alt-tab away and back again NVDA reverts to the number.
If I use an emoji in an email and send the email to myself, the emoji is presented correctly in the returned email.
This behaviour  is consistent across two computers.  Narrator is announcing emojis correctly.
Disabling add-ons makes no difference.  The result is even the same with an alpha of 2019.1. Frown!

**Comment by josephsl** [comment]
Hi, how about with different speech synthesizers (OneCore, eSpeak, etc.)? Thanks.

**Comment by andrew-l-d** [comment]
Sorry, I meant to mention that changing between eSpeak, Windows one core and SAPI 5 voices makes no difference.

 

 

Andrew

**Comment by josephsl** [comment]
Hi,

This is resolved to some degree with Word/experimental UIA support enabled.

CC @MichaelDCurran, @LeonardDer, @DKager

**Comment by Qchristensen** [comment]
No-one has mentioned which version of Office, but I'm guessing the latest updates to Office 365 (as opposed to Office 2016 or 2013 etc).  I can also repro this on Office 365, and my current version 16.0.11126.20200 (should be latest updates on regular stable update channel).

**Comment by andrew-l-d** [comment]
Apologies, I did not think of the Office version.  I had a friend who is using Office 2013 write 😊 in an email to me and it read correctly for him.  NVDA at my end did not read it correctly.  It was a few weeks ago and I did not keep a copy, but am fairly certain NVDA announced colon right parenthesis.  That is different to when I use Office 365, where I get 12 before restarting NVDA.

 

Andrew

**Comment by CyrilleB79** [comment]
Hello

As a complement with NVDA 2019.2 or latest alpha (alpha-18488,4b53f876) and word 2016 (16.0.4888.1000) :

I have tried to type the following sequence:
:-) :-| :-(
These 3 smileys are automatically changed to emoji thanks to Word auto-correction.
In legacy mode, if I read the character with Numpad 2, these 3 emojis are announced as "(" (left parenthesis = char number 0x28). Note that I remember to have them read as "12" some month ago, maybe with a different version of NVDA.
With UIA activated, they are read as "void" (no character at all).
If I copy/paste the line containing these 3 characters from Word to notepad, the 3 copied character have the following char number : 0xf04a, 0xf04b, 0xf04c. These characters belong to Private Use Area  unicode block.

@josephsl:
What did you mean by writting :
> This is resolved to some degree with Word/experimental UIA support enabled
What should we expect?

**Comment by Adriani90** [comment]
cc: @nvdaes

**Comment by josephsl** [comment]
Hi,

Six years later...

Any updates?

Tanks.

**Comment by andrew-l-d** [comment]
Emojis read well in current Word 365 with UI Automation set to Where Suitable or Always.  :) still presents as 12 if UI Automation is set to Only Where Necessary.  As Where Suitable is default, my suggestion would be to close the ticket.  However, I will let more learned people make that decision.

Andrew

**Comment by CyrilleB79** [comment]
"Where suitable" does not resolve to UIA for Windows 10 users or users of older Word versions. Until these Windows or Word versions are supported, I advise not to close this issue.

Moreover, the realty is that even if it's an advanced parameter, many users have modified this parameter to "Only where necessary", depending on their needs, since various features in UIA mode are still bugged or unavailable. I'd say that this should be discussed / taken into account before deciding to close all the Word objec model related issues.
